icu4c/source/i18n/units_data.cpp

```diff
@@ -387,24 +388,104 @@ U_I18N_API UnitPreferences::UnitPreferences(UErrorCode &status) {
     ures_getAllItemsWithFallback(unitsBundle.getAlias(), "unitPreferenceData", sink, status);
 }
 
+CharString getKeyWordValue(const Locale &locale, StringPiece kw, UErrorCode &status) {
+    CharString result;
+    CharString buffer;
+    int32_t bufferCapacity;
+    char* charBuffer = buffer.getAppendBuffer(100, 100, bufferCapacity, status);
+    int32_t kwLen = locale.getKeywordValue(kw.data(), charBuffer, bufferCapacity, status);
+  
+
+    if (U_FAILURE(status) || kwLen == 0) {
+        status = U_MISSING_RESOURCE_ERROR;
+        return result;
+    }
+
+    result.append(charBuffer, kwLen, status);
+    return result;
+}
```

Comment:
```suggestion
CharString getKeyWordValue(const Locale &locale, StringPiece kw, UErrorCode &status) {
    CharString result;
    if (U_FAILURE(status)) { return result; }
    {
        CharStringByteSink sink(&result);
        locale.getKeywordValue(kw, sink, status);
    }
    if (U_SUCCESS(status) && kw.isEmpty()) {
        status = U_MISSING_RESOURCE_ERROR;
    }
    return result;
}
```
- easier to use a sink than an output buffer
- any function that takes `UErrorCode` should first check for failure; you can defer that only if you call other functions that take `UErrorCode`
- don't clobber an error code

Reply:
thanks Markus, done!